src/mlpack/core/data/image_resize_crop.hpp

```diff
@@ -0,0 +1,140 @@
+/**
+ * @file core/data/image_resize_crop.hpp
+ * @author Omar Shrit
+ *
+ * Image resize and crop functionalities.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_CORE_DATA_IMAGE_RESIZE_CROP_HPP
+#define MLPACK_CORE_DATA_IMAGE_RESIZE_CROP_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/stb/stb.hpp>
+
+#include "image_info.hpp"
+
+namespace mlpack {
+namespace data {
+
+/**
+ * Image resize/crop interfaces.
+ */
+
+/**
+ * Resize one single image matrix or a set of images.
+ *
+ * This function should be used if the image is loaded as an armadillo matrix
+ * and the number of cols equal to the Width and the number of rows equal
+ * the Height of the image, or the total number of image pixels is equal to the
+ * number of element in an armadillo matrix.
+ *
+ * The same applies if a set of images is loaded, but all of them need to have
+ * identical dimension when loaded to this matrix.
+ *
+ * @param image The input matrix that contains the image to be resized.
+ * @param info Contains relevant input images information.
+ * @param newWidth The new requested width for the resized image.
+ * @param newHeight The new requested height for the resized image.
+ */
+template<typename eT>
+inline void ResizeImages(arma::Mat<eT>& images, data::ImageInfo& info,
+    const size_t newWidth, const size_t newHeight)
+{
+  // First check if we are resizing one image or a group of images, the check
+  // is going to be different depending on the dimension.
+  // If the user would like to resize a set of images of different dimensions,
+  // then they need to consider passing them image by image. Otherwise, as
+  // assume that all images have identical dimension and need to be resized.
+  if (images.n_cols == 1)
+  {
+    if (images.n_elem != (info.Width() * info.Height() * info.Channels()))
+    {
+      std::ostringstream oss;
+      oss << "Dimensions mismatch. ResizeImages(): the number of pixels is"
+        " not equal to the dimension provided by Info."
+        << std::endl;
+      Log::Fatal << oss.str();
+    }
+  }
+  else
+  {
+   if (images.n_rows != (info.Width() * info.Height() * info.Channels()))
+    {
+      std::ostringstream oss;
+      oss << "Dimensions mismatch. ResizeImages(): In the case of several"
+        " images, please check if all the images have the same dimensions"
+        " already, if not, load each image in one column and recall this"
+        " function."
+        << std::endl;
+      Log::Fatal << oss.str();
```

Comment:
Actually you don't need the `ostringstream`, you can output directly to `Log::Fatal`.  (The same comment applies above too.)